### PR TITLE
Override: Support test names

### LINKED ISSUE #21556: prow: override: use test names, not status contexts
Today, users interact with Prow jobs by using `/test <name>`. However, administrators must use `/override <context>` even when the context is not the test name. We should allow `/override` to use test names as that's what users are familiar with.

/cc @bparees 

WDYT @cjwagner @chaodaiG @alvaroaleman ?

### PR DESCRIPTION
**Pull request by YuviGold**
Resolves #21556

 `/override <token>`
Will search:

- job names
- contexts on the PR

### CODE REVIEW

File: prow/plugins/override/override.go [hunk 1]
```diff
@@ -334,15 +335,31 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent,
...
+
 	contexts := sets.NewString()
 	for _, status := range statuses {
 		if status.State == github.StatusSuccess {
 			continue
 		}
+
 		contexts.Insert(status.Context)
+
+		if result := funk.Find(presubmits, func(job config.Presubmit) bool {
```
Comment (alvaroaleman) [inline prow/plugins/override/override.go:354]: Iterating over a slice to find a matching result is absolutely not worth an external dependency

File: prow/plugins/override/override.go [hunk 2]
```diff
@@ -334,15 +335,31 @@ func handle(oc overrideClient, log *logrus.Entry, e *github.GenericCommentEvent,
...
+			return job.Context == status.Context
+		}); result != nil {
+			contexts.Insert(result.(config.Presubmit).Name)
+		}
 	}
+
 	if unknown := overrides.Difference(contexts); unknown.Len() > 0 {
-		resp := fmt.Sprintf(`/override requires a failed status context to operate on.
+		resp := fmt.Sprintf(`/override requires a failed status context or a Prow Job name to operate on.
```
Comment (alvaroaleman) [inline prow/plugins/override/override.go:362]:
```suggestion
		resp := fmt.Sprintf(`/override requires a failed status context or a job name to operate on.
```

Prowjobs are the custom resource instances and they have a randomized name so this is slightly confusing right now

File: prow/plugins/override/override_test.go [hunk 3]
```diff
@@ -587,6 +600,93 @@ func TestHandle(t *testing.T) {
...
+				"ci/prow/prow-job": {
+					Context:     "ci/prow/prow-job",
+					State:       github.StatusSuccess,
+					Description: description(adminUser),
+				},
+			},
+		},
+		{
+			name:    "override same context and prow job",
+			comment: "/override ci/prow/prow-job\n/override prow-job",
```
Comment (alvaroaleman) [inline prow/plugins/override/override_test.go:663]: This test only differs in the order of the comment from the previous one?

File: prow/plugins/override/override_test.go [hunk 4]
```diff
@@ -587,6 +600,93 @@ func TestHandle(t *testing.T) {
...
+					State:       github.StatusFailure,
+				},
+			},
+			presubmits: map[string]config.Presubmit{
+				"prow-job": {
+					JobBase: config.JobBase{
+						Name: "prow-job",
+					},
+					Reporter: config.Reporter{
+						Context: "ci/prow/prow-job",
```
Comment (alvaroaleman) [inline prow/plugins/override/override_test.go:648]: This test would already have passed before your change. Lets do this with a presubmit whose context is not the same as the one in the override command

File: prow/plugins/override/override_test.go (hunk 3 above)
Comment (YuviGold) [reply prow/plugins/override/override_test.go:663]: Yes. I'll make the name clearer.

File: prow/plugins/override/override_test.go (hunk 4 above)
Comment (YuviGold) [reply prow/plugins/override/override_test.go:648]: The test would have failed before it wouldn't accept "prow-job" as a valid override.
It was created mostly to verify both prow-job and the context names can be used together. But I'll change it as you asked

**Review by fejta — APPROVED** [review]
Great idea, thanks!

/hold

Alvaro or Cole any more comments?

**Review by alvaroaleman — APPROVED** [review]
looks good, thanks!
/hold cancel

**Pull request merged**